### PR TITLE
fix(TDI-44393) : Support schema date pattern.

### PR DESCRIPTION
**Pull request by ypiel-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-44393
Studio schema date pattern property were not used by tcompV0/Snowflake connector.

**What is the new behavior?**
If we receive a Date as data and DBType output is String/Text/Varchar, then we format the date to String using the given date pattern.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/428/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/428/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/455/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/455/)

File: components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/tsnowflakeoutput/TSnowflakeOutputProperties.java [hunk 1]
```diff
@@ -83,6 +83,12 @@
      */
     public Property<DateMapping> dateMapping = newEnum("dateMapping", DateMapping.class);
 
+    /**
+     * Advanced property to indicate if we use default snowflake driveer date -> string conversion,
```
Comment (Voievidko) [inline components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/tsnowflakeoutput/TSnowflakeOutputProperties.java:87]:
```suggestion
     * Advanced property to indicate if we use default snowflake driver date -> string conversion,
```

Comment (ypiel-talend) [reply components/components-snowflake/components-snowflake-definition/src/main/java/org/talend/components/snowflake/tsnowflakeoutput/TSnowflakeOutputProperties.java]: fixed

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/458/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/458/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/463/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/463/)

**Pull request merged**